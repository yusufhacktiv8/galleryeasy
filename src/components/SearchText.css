.search-text {
  border: 0;
  border-bottom: 1px solid silver;
  font-size: 26px;
  color: gray;
  width: 100%;
  outline-width: 0;
}
