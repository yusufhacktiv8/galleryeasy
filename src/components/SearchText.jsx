import React, { Component } from 'react';
import './SearchText.css';

export default class SearchText extends Component {
  state = {
    search: '',
  }

  onSearchChange = (e) => {
    this.setState({
      search: e.target.value,
    });
  }

  onKeyPress = (e) => {
    if (e.key === 'Enter') {
      this.props.onSearch(this.state.search);
    }
  }

  render() {
    return (
      <input
        type="text"
        value={this.state.search}
        onChange={this.onSearchChange}
        onKeyPress={this.onKeyPress}
        className="search-text"
        placeholder="Start searching for images!"
      />
    );
  }
}
