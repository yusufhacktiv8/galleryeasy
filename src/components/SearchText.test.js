import React from 'react';
import renderer from 'react-test-renderer';
import SearchText from './SearchText';

test('SearchText correctly rendered', () => {
  const component = renderer.create(
    <SearchText />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
