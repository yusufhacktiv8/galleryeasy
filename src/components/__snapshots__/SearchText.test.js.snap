// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchText correctly rendered 1`] = `
<input
  className="search-text"
  onChange={[Function]}
  onKeyPress={[Function]}
  placeholder="Start searching for images!"
  type="text"
  value=""
/>
`;
